apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/AbstractR4ResourceProvider.java

```diff
@@ -151,4 +157,97 @@ public T read(@IdParam IdType claimId, RequestDetails requestDetails) {
    */
   @VisibleForTesting
   abstract Optional<ResourceTypeV2<T>> parseClaimType(String typeText);
+
+  /**
+   * Returns a set of all supported resource types.
+   *
+   * @return Set of all supported resource types.
+   */
+  @VisibleForTesting
+  abstract Set<ResourceTypeV2<T>> getResourceTypes();
+
+  @Search
+  @Trace
+  public Bundle findByPatient(
+      @RequiredParam(name = "mbi")
+          @Description(shortDefinition = "The patient identifier to search for")
+          ReferenceParam mbi,
+      @OptionalParam(name = "type")
+          @Description(shortDefinition = "The source type to use for fetching claim data")
+          String type,
+      @OptionalParam(name = "isHashed")
+          @Description(shortDefinition = "A boolean indicating whether or not the MBI is hashed")
+          String hashed,
+      @OptionalParam(name = "_lastUpdated")
+          @Description(shortDefinition = "Include resources last updated in the given range")
+          DateRangeParam lastUpdated,
+      @OptionalParam(name = "service-date")
+          @Description(shortDefinition = "Include resources that completed in the given range")
+          DateRangeParam serviceDate,
+      RequestDetails requestDetails) {
+    if (mbi != null && !StringUtils.isBlank(mbi.getIdPart())) {
+      String mbiString = mbi.getIdPart();
+
+      Bundle bundleResource;
+
+      boolean isHashed = !Boolean.FALSE.toString().equalsIgnoreCase(hashed);
+
+      Optional<ResourceTypeV2<T>> optional = parseClaimType(type);
+      if (optional.isPresent()) {
+        bundleResource =
+            createBundleFor(
+                Collections.singleton(optional.get()), mbiString, isHashed, lastUpdated);
```

Comment:
what happens if the user specifies an invalid query parameter?  For example, `type=OUTPATIENT,INPATIENT`?